sdk/keyvault/azure-keyvault-keys/CHANGELOG.md

```diff
@@ -9,6 +9,8 @@
 ### Bugs Fixed
 
 ### Other Changes
+- The `KeyReleasePolicy` constructor raises a useful `TypeError` if the provided `encoded_policy` has an incorrect type
+  ([#30000](https://github.com/Azure/azure-sdk-for-python/issues/30000))
```

Comment:
If a string was given, the old behavior is that the `KeyReleasePolicy` would be successfully created, with a string `encoded_policy` property. When the `KeyReleasePolicy` instance was passed to a method like `create_key`, though -- like in https://github.com/Azure/azure-sdk-for-python/issues/29945 -- the service would raise an error suggesting that no release policy was provided

So to your point, this change is breaking in the sense that the policy can no longer be successfully created with a string, but it's not breaking in the sense that an incorrectly created policy would have always been unusable for service requests

Reply:
...Which means that maybe a less-breaking change would be to evaluate the `encoded_policy` type when the `KeyReleasePolicy` is _used_, rather than when it's created... hmm...

Reply:
So we are changing a runtime AzureError into a ctoring TypeError?

Given the scenario is broken now, I think it is acceptable.

@annatisch @johanste to confirm.